src/support/variables/typography.scss

```diff
@@ -21,8 +21,8 @@ $h6-size: 12px !default;
 $font-size-small: 12px !default;
 
 // Font weights
-$font-weight-bold: 600 !default;
-$font-weight-semibold: 500 !default;
+$font-weight-bold: 700 !default;
```

Comment:
Last quarter we normalized `font-weight` [with primitives ](https://primer.style/primitives/typography#weight)(this has yet to be fully embedded into Primer CSS) to align with OpenType, but we actually cut `bold` as an option.

`semibold` being 600 will replace instances of the old `bold` to remove any risk of a breaking visual change. For `500` we added `medium`. Since this PR is focused on marketing, and there are already custom weight variables being used it might make sense to remove these more global changes for now so we can follow up with the primitives rollout.

Alternatively, we could do something like...

```
$font-weight-bold: $font-weight-semibold;
$font-weight-semibold: 600;
$font-weight-medium: 500;
$font-weight-normal: 400;
$font-weight-light: 300;
```

This would still impact `semibold`, but I think that gets less use than `bold` right now and maybe a less dramatic change? But might still makes sense to try and scope this just to marketing for now.

Reply:
@langermank Oooh, interesting!

## Scoping to marketing

If we go down route we would have to define the `bold` utility again inside of the marketing bundle, and if you then include the marketing bundle you would either 1) change how `text-bold` looks *everywhere*, or 2) we scope it to only override `text-bold` within a certain marketing context (e.g. `.font-mktg`, like we do elsewhere). But the latter feels a bit overly complex, and the system becomes inconsistent in how it behaves.

## Deprecating `bold` outside of marketing, replacing `bold` with `semibold`, and `semibold` with `medium`

But given that we can't really cleanly separate these two problem spaces (product and marketing), and that we've already made the decision to deprecate `bold` and move to align with the OpenType spec without any visual changes… shouldn't we just go down this route now rather than later? Or are we waiting to time it with something in particular? I feel like we would be able to replace all instances of `text-semibold` with `text-medium` and all instances of `text-bold` with `text-semibold` quite easily? 🙏

Reply:
@tobiasahlin 

> shouldn't we just go down this route now rather than later? Or are we waiting to time it with something in particular?

The only reason we haven't made these updates yet is resourcing! 😄 if you have the bandwidth to make these changes in PCSS and also roll it out to dotcom (with our review support) then I'd love to see this happen!

Are you thinking we'll totally deprecate `bold` variables and corresponding utility classes? I think the utility is where it gets tricky, since we'll need to do a massive find/replace in dotcom. What do you think? I'm happy to support.

Reply:
@langermank Yesss, I'd love to help with rolling that out, and lead with your support and oversight 🙏 ✨ 

My suggestion would be to completely deprecate `bold` (vars and utilities) for now, to make sure that we can re-introduce it with the correct value later, if we ever feel that we want to 👀 what do you think?

Reply:
@tobiasahlin I love that idea! I think we might have a deprecation strategy but I've never done it 🤔 maybe @jonrohan has some insight to share. Even if we bump everything in dotcom ourselves, we might still need to document the deprecation somehow. Let me know how I can help with this!

Reply:
@langermank Wonderful! ✨ Let me have a go to prepare for that in this PR 🙏